position/sqrt_price_math.gno

```diff
@@ -80,6 +80,34 @@ func sqrtPriceMathGetNextSqrtPriceFromAmount1RoundingDown(
 		quotient = new(u256.Uint).Sub(quotient, u256.One())
 		return new(u256.Uint).Sub(sqrtPX96, quotient)
 	}
+
+	// quotient mutst be positive when amount and liquidity are positive
+	if add {
+		if amount.Lte(u256.UnsafeFromDecimal(consts.MAX_UINT160)) {
+			quotient = new(u256.Uint).Lsh(amount, 96)
+			quotient = new(u256.Uint).Div(quotient, liquidity)
+		} else {
+			quotient = u256.MulDiv(amount, u256.UnsafeFromDecimal(consts.Q96), liquidity)
+		}
```

Comment:
It would be better to extract the part that calculates the quotient into a function. Also, since both the if and else branches return some value, the code in the else block could be pulled out.

Reply:
Oh yes, if my memory serves me right `sqrt_price_math` was handling it with separate function when we're using `bigint`.

But for now we uses `u256/i256` and I had to roll back in old ways if test cases are find

Reply:
https://github.com/gnoswap-labs/gnoswap/pull/195/commits/7ef5241f1d962daf3125008f9206a442d37b2adf